### ISSUE TITLE
Investigate using `sv_fps 40` as a default

### ISSUE THREAD

**Issue by Calinou** [etlegacy/etlegacy#2555]
A lot of effort has been put into fixing `sv_fps`-related issues, so much that most competitive gameplay is now done with `sv_fps 40`. Unfortunately, most casual players won't get to experience this increased tickrate, despite it being beneficial at all skill levels.

There are many advantages to running a higher tickrate on public servers or local matches:

- Other players' movement follow more closely the player's actual position. Even though there's lag compensation and animation-induced weapon firing delay has been resolved recently (thanks again!), there's still a noticeable delay between clicking the left mouse button and hitting a player due to the tick rate. This also reduces peeker's advantage by design.
- Crosshair spread is updated more smoothly. Right now, it's not interpolated, so the 20 Hz update rate is rather obvious.
  - The crosshair spread display could be interpolated on the client side, but this would also increase the perceived delay between the spread update (which isn't predicted on the client-side) and the displayed value. This might be more viable with 40 Hz tickrate, as the delay will be less problematic.
- Being pushed around by other players or moving vehicles is smoother (or at least less frustrating :stuck_out_tongue:).
- Local bot matches feel smoother, with less delay when firing weapons.
- Spectating a player looks more accurate to their true actions.

In terms of CPU/bandwidth requirements, I don't know if it's been tinkered with recently, but I'd expect most public servers to be able to run `sv_fps 40` without trouble.

It might be challenging for the largest active servers like TeamMuppet though (with their 48-player + 16 spectator limit), so perhaps the tickrate should be decided depending on player count (with `sv_fps` being adjusted every few minutes to account for the current player count). Adjusting `sv_fps` during gameplay will cause a small stutter, so we should ensure this isn't done during the heat of the action (e.g. in the last 2 minutes of a match). It should preferably done when many players are wounded or dead, so that fewer players will notice the tickrate change kicking in.

For instance, this could be controlled with the following cvars:

```c
// Default FPS (used on server startup and restored after a long enough duration below the threshold).
sv_fps 40

// Maximum amount of players where the default FPS value is used.
// This value would support up to 12v12 matches (ignoring spectators).
sv_fpsMaxPlayers 24

// `sv_fps` fallback value when above player threshold.
sv_fpsAboveMaxPlayers 20
```

This is pending on https://github.com/etlegacy/etlegacy/issues/1637 to be fully resolved, but from what I've seen, most `sv_fps`-related issues either seem to be resolved, or are hard enough to notice at 40 Hz.

Note that if we carry over this change, it might be worth using `cl_maxpackets 125` as a default as well to further improve smoothness and reduce jitter (the current default seems to be `100`).

See [my comment over on Xonotic's GitLab](https://gitlab.com/xonotic/xonotic-data.pk3dir/-/issues/2860#note_1632789686) for more detailed rationale. There are already at least 2 id Tech 3-based games defaulting to 40 Hz (Quake Live and Unvanquished) - let's try to make ET:Legacy the third one :slightly_smiling_face:

**Comment by u1ps** [comment]
TeamMuppet with a lot of players is a challenge indeed but servers in range of recommended player count should not run into much problems with correctly set up server.

40 snaps is superior in many ways mentioned above and should be default if there are not any known or unknown critical issues similar to https://github.com/etlegacy/etlegacy/issues/1637

However swap of snaps based on player count is bad idea. It changes the game feel and if u are used to playing with responsive 40 and need to swap between 20 vs 40 behavior it is too big change from player perspective how things work and run ( Muscle memory, Where u have to aim on ur screen to get the hits y etc.) Unstable gaming experience where next game can feel different due 25ms or 50ms server frame.

**Comment by pepelevamp** [comment]
I agree. Ultimately you want the other players actions to be true and up to date.  And everything to overall be more responsive. With a higher server fps, any artifacts it introduces (even say more accurate representation of other-player network jitter ) are arguably things that should be fixed via means other than slowing & splining the responsiveness.

50ms is pretty big, even for predicted and splined animation.

**Comment by Aciz** [comment]
Something to keep in mind is that both `sv_fps` and `snaps` are engine cvars, and while having higher `snaps` by default doesn't hurt anything (if you set it to higher than `sv_fps`, you just don't get any benefits from it as the server ultimately dictates the update rate), making `sv_fps 40` the default will break all mods besides legacy and etjump, as those are the only mods which have resolved all the issues that raising `sv_fps` causes. So realistically, you have to make it so that the new default only applies to legacy mod (etjump doesn't really care as we run `sv_fps 125` anyway, which is not really useful for frag mods).

**Comment by pepelevamp** [comment]
Oh really.

Fair call. I freely admit I am unaware of the implications.

I have to say though it does look pretty flawless to me. If you didn't tell
me it was 50ms I never would have known.
